targets/mbedos5/jerryscript-mbed/jerryscript-mbed-drivers/source/setTimeout-js.cpp

```diff
@@ -28,11 +28,20 @@ DECLARE_GLOBAL_FUNCTION(setTimeout) {
     CHECK_ARGUMENT_TYPE_ALWAYS(global, setTimeout, 0, function);
     CHECK_ARGUMENT_TYPE_ALWAYS(global, setTimeout, 1, number);
 
-    jerry_acquire_value(args[0]);
     int interval = int(jerry_get_number_value(args[1]));
 
     int id = mbed::js::EventLoop::getInstance().getQueue().call_in(interval, jerry_call_function, args[0], jerry_create_null(), (jerry_value_t*)NULL, 0);
 
+    jerry_value_t result = jerry_set_property_by_index(this_obj, id, args[0]);
```

Comment:
Same here, might be better not polluting the global object.

And also the question about dropping the reference: property should be cleared when timeout in explicitly canceled (via `clearTimeout`) _but also_ if/after the event fired (because this is a non-repeating one-shot event, if I'm right). (I'm not sure whether the second scenario can be handled easily, because I don't know whether we get notified about that anywhere and/or whether we still have reference to `function_obj_p` and `id` there.)